addOns/spider/src/main/java/org/zaproxy/addon/spider/ExtensionSpider2.java

```diff
@@ -19,37 +19,162 @@
+
+    /** The options spider panel. */
+    private OptionsSpiderPanel optionsSpiderPanel = null;
+
+    /** The params for the spider. */
+    private SpiderParam params = null;
+
+    private List<SpiderParser> customParsers;
+    private List<FetchFilter> customFetchFilters;
+    private List<ParseFilter> customParseFilters;
+
+    private SpiderAPI spiderApi;
+
+    private SpiderScanController scanController = null;
+
+    private Icon icon;
+
+    private boolean panelSwitch = true;
+
+    /**
+     * The list of excluded patterns of sites. Patterns are added here with the ExcludeFromSpider
+     * Popup Menu.
+     */
+    private List<String> excludeList = Collections.emptyList();
+
+    private ZapMenuItem menuItemCustomScan = null;
 
     public ExtensionSpider2() {
         super(NAME);
+        initialize();
+    }
+
+    /** This method initializes this extension. */
+    private void initialize() {
+        this.setOrder(EXTENSION_ORDER);
+        this.customParsers = new LinkedList<>();
+        this.customFetchFilters = new LinkedList<>();
+        this.customParseFilters = new LinkedList<>();
+        this.scanController = new SpiderScanController(this);
+    }
+
+    public void setValueGenerator(ValueGenerator generator) {
+        if (generator == null) {
+            throw new IllegalArgumentException("Parameter generator must not be null.");
+        }
+        this.generator = generator;
+    }
+
+    public ValueGenerator getValueGenerator() {
+        return generator;
     }
 
     @Override
     public void hook(ExtensionHook extensionHook) {
-        ExtensionSpider spider = getExtensionSpider();
-        svgHrefSpider = new SvgHrefSpider();
-        spider.addCustomParser(svgHrefSpider);
-        LOGGER.debug("Added custom SVG HREF spider.");
+        boolean isCoreSpiderDisabled =
+                ExtensionSpider.class.getAnnotation(Deprecated.class) != null;
+        if (!isCoreSpiderDisabled) {
+            ExtensionSpider2 spider = getExtensionSpider();
+            svgHrefSpider = new SvgHrefSpider();
+            spider.addCustomParser(svgHrefSpider);
+            LOGGER.debug("Added custom SVG HREF spider to core spider.");
+        } else {
+            // Register for listeners
+            extensionHook.addSessionListener(this);
+
+            // Initialize views
+            if (getView() != null) {
+                extensionHook.getHookMenu().addToolsMenuItem(getMenuItemCustomScan());
+                extensionHook.getHookView().addStatusPanel(getSpiderPanel());
+                extensionHook.getHookView().addOptionPanel(getOptionsSpiderPanel());
+                extensionHook.getHookMenu().addPopupMenuItem(getPopupMenuItemSpiderDialog());
+                extensionHook
+                        .getHookMenu()
+                        .addPopupMenuItem(getPopupMenuItemSpiderDialogWithContext());
+                ExtensionHelp.enableHelpKey(getSpiderPanel(), "ui.tabs.spider");
```

Comment:
Will have to be updated, fine if just in the help migration commit (there might be other places, e.g. options panel).

Reply:
I've updated this and two others in the help commit. However, for some reason tab and options aren't working. The base element and dialog seem fine though.

Reply:
Also links from the migrated help, to core items no longer work. I'm not sure if this is because I have the traversals wrong now or if it's like trying to link between add-ons.

Reply:
Links between helpsets (add-ons and core) do not work. Those will have to be removed.

Reply:
For the help not working in the tab and options, will have to look.

Reply:
To be clear when I said tab and options weren’t working, I meant within the help itself, not on the panels.